new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/KubernetesExec.java

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.weblogic.kubernetes.actions.impl.primitive;
+
+import java.io.IOException;
+
+import io.kubernetes.client.Exec;
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.models.V1Pod;
+
+/**
+ * KubernetesExec object for executing a command in the container of a pod.
+ */
+public class KubernetesExec {
+  // the Kubernetes api client to dispatch the "exec" command
+  private ApiClient apiClient;
+
+  // The Kubernetes pod where the command is to be run
+  private V1Pod pod;
+
+  // the container in which the command is to be run
+  private String containerName;
+
+  // If true, pass a stdin stream into the container
+  private boolean passStdinAsStream;
+
+  // If true, stdin is a TTY (only applies if stdin is true)
+  private boolean stdinIsTty;
+
+  public KubernetesExec apiClient(ApiClient apiClient) {
+    this.apiClient = apiClient;
+    return this;
+  }
+
+  /**
+   * Get the API client for these exec operations.
+   *
+   * @return The API client that will be used.
+   */
+  public ApiClient getApiClient() {
+    return apiClient;
+  }
+
+  public void setApiClient(ApiClient apiClient) {
+    this.apiClient = apiClient;
+  }
+
+  public KubernetesExec pod(V1Pod pod) {
+    this.pod = pod;
+    return this;
+  }
+
+  /**
+   * Get the Kubernetes pod where the command is to be run.
+   *
+   * @return Kubernetes pod object
+   */
+  public V1Pod getPod() {
+    return pod;
+  }
+
+  public void setPod(V1Pod pod) {
```

Comment:
Why do we need multiple setters for "pod"?  And, are we doing fluent API or not?  This class seems to mix both.  For example line 50 and 64.

Reply:
yes, it is a mix.  I was looking at the K8s api resource objects and our model objects and fashioned it after them but realized that they have set/get methods in order  to be serialized/de-serialized using the JSON/YAML libraries.  I can clean it up to have the more fluent api look.  Thanks

Reply:
Removed set/get methods since not necessary.